From 70463cf1f752f73c75970d3a4b94039fe9a49db4 Mon Sep 17 00:00:00 2001
From: Costin Lupu <costin.lupu@cs.pub.ro>
Date: Mon, 30 Jan 2023 21:03:56 +0100
Subject: [PATCH] cloning: profiling

---
 src/rdb.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/src/rdb.c b/src/rdb.c
index acbba4b..2c6af6a 100644
--- a/src/rdb.c
+++ b/src/rdb.c
@@ -32,6 +32,7 @@
 #include "zipmap.h"
 #include "endianconv.h"
 #include "stream.h"
+#include "profile.h"
 
 #include <math.h>
 #include <sys/types.h>
@@ -1326,6 +1327,7 @@ werr:
 int rdbSaveBackground(char *filename, rdbSaveInfo *rsi) {
     pid_t childpid;
     long long start;
+    const char *label;
 
     if (server.aof_child_pid != -1 || server.rdb_child_pid != -1) return C_ERR;
 
@@ -1334,13 +1336,23 @@ int rdbSaveBackground(char *filename, rdbSaveInfo *rsi) {
     openChildInfoPipe();
 
     start = ustime();
-    if ((childpid = fork()) == 0) {
+
+    PROFILE_NESTED_TICK();
+    childpid = fork();
+    label = (childpid == 0) ? "fork child" : "fork parent";
+    PROFILE_NESTED_TOCK_MSEC(label);
+
+
+    if (childpid == 0) {
         int retval;
 
+        PROFILE_NESTED_TICK();
         /* Child */
         closeListeningSockets(0);
         redisSetProcTitle("redis-rdb-bgsave");
+        PROFILE_NESTED_TICK();
         retval = rdbSave(filename,rsi);
+        PROFILE_NESTED_TOCK_MSEC("save");
         if (retval == C_OK) {
             size_t private_dirty = zmalloc_get_private_dirty(-1);
 
@@ -1353,6 +1365,7 @@ int rdbSaveBackground(char *filename, rdbSaveInfo *rsi) {
             server.child_info_data.cow_size = private_dirty;
             sendChildInfo(CHILD_INFO_TYPE_RDB);
         }
+        PROFILE_NESTED_TOCK_MSEC("total");
         exitFromChild((retval == C_OK) ? 0 : 1);
     } else {
         /* Parent */
-- 
2.20.1

